### DISCUSSION TITLE
EQ3 communication fails "FAILCONNECT"

### DISCUSSION THREAD

**Discussion by dog-rgb** [arendst/Tasmota#14313] [Q&A] [answered]
Hi, I am using two ESP32 devices with tasmota32-bluetooth.bin, version 2022.01.1. So the EQ3 communication is enabled by default. One device is running fine, the second one is reporting usually "FAILCONNECT" when an EQ3 device is polled:
```
00:35:17.864 EQ3 EQ3_14A_Wohnzimmer: Op queued len now 1
00:35:17.870 EQ3 EQ3_14A_Wohnzimmer:Op dequeued len now 0
00:35:33.858 MQT: tele/tasmota_5DEF14/BLE = {"BLEOperation":{"opid":"174","stat":"-11","state":"FAILCONNECT","MAC":"001A2211D531","u":"128","svc":"3e135142-654f-9090-134a-a6ff5bb77046","char":"3fa4585a-ce4a-3bad-db4b-b8df8179ea09","notifychar":"d0e8434d-cd29-0996-af41-6c90f4e0eb2a","write":"03160107002311000000000000000000"}}
00:35:42.836 MQT: tele/tasmota_5DEF14/BLE = {"BLEOperation":{"opid":"175","stat":"-11","state":"FAILCONNECT","MAC":"001A2211D531","u":"128","svc":"3e135142-654f-9090-134a-a6ff5bb77046","char":"3fa4585a-ce4a-3bad-db4b-b8df8179ea09","notifychar":"d0e8434d-cd29-0996-af41-6c90f4e0eb2a","write":"03160107002311000000000000000000"}}
00:35:58.879 MQT: tele/tasmota_5DEF14/BLE = {"BLEOperation":{"opid":"176","stat":"-11","state":"FAILCONNECT","MAC":"001A2211D531","u":"128","svc":"3e135142-654f-9090-134a-a6ff5bb77046","char":"3fa4585a-ce4a-3bad-db4b-b8df8179ea09","notifychar":"d0e8434d-cd29-0996-af41-6c90f4e0eb2a","write":"03160107002311000000000000000000"}}
00:36:09.834 EQ3 001A2211D531 poll fail not sent because EQ3HideFailedPoll
00:36:09.851 MQT: tele/tasmota_5DEF14/BLE = {"BLEOperation":{"opid":"177","stat":"-11","state":"FAILCONNECT","MAC":"001A2211D531","u":"128","svc":"3e135142-654f-9090-134a-a6ff5bb77046","char":"3fa4585a-ce4a-3bad-db4b-b8df8179ea09","notifychar":"d0e8434d-cd29-0996-af41-6c90f4e0eb2a","write":"03160107002311000000000000000000"}}
```
What can be the reason?

Any additional information needed?

**Comment by barbudor** [comment, marked as answer]
Not using those but just a guess : if the BLE driver works in connected mode with the EQ3, I would assume that the EQ3 can be connected to only 1 BLE host
Sometime such devices have a bluetooth icon on their LCD to provide status of bluetooth connection...

May be you can join the Discord server in channel #ble-esp32. There, I know that @ Ysbrand @ kroonen @ tony are using such thermostat.

**Reply by dog-rgb** [reply]
This sounds reasonable. Meanwhile I also found the parameter TRVonlyaliased to restrict the access to specific EQ3 devices (I am using the Alias already to get the EQ3 information from specific BLE devices only), so I am going to reset the bluetooth connections on those EQ3 devices. If that doesn't help I will check the communication you proposed.

**Reply by Ysbrand** [reply]
No, the EQ3 can communicate with multiple ESP32 units. Obviously not at exactly the same time, and you might see some BLE interference from time to time. I suspect that there is an issue with the ESP32, maybe relocate it? But could be a simple HW failure as well.

**Comment by InspectorColumbo** [comment]
I think newer EQ3 Firmware need pairing. But Tasmota dosent support pairing. That ist why the connection fails.

**Comment by Ysbrand** [comment]
What FW version are you running? 1.2 and 1.46 both have been reported as running ok without pairing.
